sunpy/map/mapbase.py

```diff
@@ -855,15 +855,17 @@ def pixel_to_world(self, x, y, origin=0):
             A coordinate object representing the output coordinate.
 
         """
-        x, y = self.wcs.wcs_pix2world(x, y, origin)
 
-        # If the wcs is celestial it is output in degress
-        if self.wcs.is_celestial:
-            x = u.Quantity(x, u.deg)
-            y = u.Quantity(y, u.deg)
-        else:
-            x = u.Quantity(x, self.spatial_units[0])
-            y = u.Quantity(y, self.spatial_units[1])
+        # Hold the WCS instance here so we can inspect the output units after
+        # the pix2world call
+        temp_wcs = self.wcs
```

Comment:
Basically we don't want to change the actual WCS instance?

Reply:
There is no 'actual wcs instance' it's a property that makes one on access.

Reply:
To-mato, tom-ato. 

So this way, you do not change the property?

Reply:
We need to do the world2pix call and then inspect cunit on the same instance. If we re-accessed `self.wcs` it wouldn't be the same instance.

Reply:
fair enough